### PR TITLE
cleanup - code conventions

### PR DESCRIPTION
**Pull request by loechel**
make diazo more readable and cleanup some things

### CODE REVIEW

**Comment by loechel** [comment]
base pull request for maybe following pulls.

Ready to merge, but no need for a release soon.

File: tox.ini [hunk 1]
```diff
@@ -4,11 +4,115 @@
...
+setenv =
+    COVERAGE_FILE=.coverage.{envname}
+
+deps =
+    pdbpp
+    pytest
+    pytest-cov
+    pytest-remove-stale-bytecode
+    pytest-mock
+    pytest-html
```
Comment (gforcada) [inline tox.ini]: Would it be possible to move these (and all the other dependencies) to a requirements.txt file (one per each different environment inside a ``requirements`` folder)? see http://tox.readthedocs.io/en/latest/example/basic.html#depending-on-requirements-txt

This way we can pin down the exact versions and have a reproducible CI environment to run tox.

File: tox.ini [hunk 2]
```diff
@@ -4,11 +4,115 @@
...
+    coverage combine
+    coverage report
+    coverage html
+    coverage xml
+
+[testenv:isort]
+basepython = python2.7
+deps = isort
+commands =
+    #isort --check-only --recursive {toxinidir}/lib
```
Comment (gforcada) [inline tox.ini]: There are a few commands commented out here and there on this file, is this a temporal status or a leftover? (after that much work!!)

File: tox.ini [hunk 3]
```diff
@@ -4,11 +4,115 @@
...
+    flake8-html
+    flake8-blind-except
+    flake8-coding
+    flake8-commas
+    flake8-debugger
+    flake8-deprecated
+    flake8-isort
+    flake8-pep3101
+#    flake8-plone-api
+#    flake8-plone-hasattr
```
Comment (gforcada) [inline tox.ini]: Actually I should have renamed this to flake8-hasattr as it is no plone specific, but a generic python one: hasattr swallows exceptions and that is usually what you don't want, usually of course.

File: tox.ini (hunk 1 above)
Comment (loechel) [inline tox.ini]: @gforcada yes that would be possible, but I don't like that actually, cause I want readable configurations.

perfect would be something like:

    deps = 
        -c https://jenkins.plone.org/test-resources/constrains.txt
        pytest
        ...

and that should actually work, as tox just pass all of the dependency parameters to pip

File: tox.ini (hunk 3 above)
Comment (loechel) [inline tox.ini]: that would be nice, if you could rename it.

File: tox.ini (hunk 2 above)
Comment (loechel) [inline tox.ini]: some are left overs, some are a kind of todo marker for me, I would like to move the files from lib to src, as that is more straight forward. It also would cause less trouble with virtualenvs.

Comment (gforcada) [inline tox.ini]: Do that on another pull request though. So we can ensure nothing breaks and commits are easy to follow for us or anyone in 3/4/6 months/years.

File: tox.ini (hunk 1 above)
Comment (gforcada) [inline tox.ini]: So we really can pass a constraints.txt to tox? Cool, as the dependencies are not really Plone specific, keep it on this repo, that's going to be less brittle

File: tox.ini (hunk 3 above)
Comment (gforcada) [inline tox.ini]: Yeah at some point I will try, not now though.

File: tox.ini (hunk 1 above)
Comment (loechel) [inline tox.ini]: seams to work, I have added a basic constrains.txt to this repository.

**Comment by loechel** [comment]
@gforcada so now we have a constrains.txt in this repository which is used in tox and travis-ci.

should be working now and ready to merge

**Comment by pbauer** [comment]
@loechel diazo 1.2.6 breaks the coredev with 

    IOError: [Errno 2] No such file or directory: '/home/jenkins/.buildout/eggs/diazo-1.2.6-py2.7.egg/diazo/update-namespace.xsl'

Some files (like `update-namespace.xsl`) are in the repo but not in the egg. See http://jenkins.plone.org/job/plone-5.1-python-2.7/433/consoleText

**Comment by pbauer** [comment]
Fixed with https://github.com/plone/diazo/commit/a1fbf3d482172474c347840f7276cd1586aaab9a
